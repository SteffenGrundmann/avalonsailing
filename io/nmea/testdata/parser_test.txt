# Tests the parsing of NMEA sentences (only actually splitting and storing
# arguments, testing the checksum, not interpreting the data).
# GPS:
$GPRMC,123519,A,4807.038,N,01131.000,E,022.4,084.4,230394,003.1,W*6A
$GPRMC,123519,A,4807.038,S,01131.000,E,022.4,084.4,230394,003.1,W*77
$GPRMC,181704,A,5209.6815,N,00643.0724,E,000.1,187.0,310109,,*17
# Wind:
$WIMWV,214.8,R,0.1,K,A*28
$IIMWV,273,R,24.3,N,A*10
# Next line is correctly parsed, but will later not be identified as a GPS
# sentence (extra fields).
$GPRMC,123519,A,4807.038,S,02121.000,W,015.4,084.4,230394,003.1,foobar,bazbladata*09
# Unknown type. Parsed correctly, but later not identified as a known type.
$FOO,123,bar,bla,baz
# Error line (> 160 chars, too much data).
$GPRMC,123519,A,4807.038,S,02121.000,W,015.4,084.4,230394,003.1,foobar,bazbladataaaaaaaaaaaaaaaaaaaaaaaaaaaaaaauuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuuababababababab*09
# Error line (> 32 args, too many args).
$GPGSV,2,1,08,01,40,083,46,02,17,308,41,12,07,344,39,14,22,228,45,308,41,12,07,344,39,14,22,228,45,39,14,22,228,45*75
# Error line, incorrect checksum.
$GPRMC,123519,A,4807.038,N,01131.000,E,022.4,084.4,230394,003.1,W*6B
