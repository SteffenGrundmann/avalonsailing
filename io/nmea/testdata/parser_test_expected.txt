Sentence parsed correctly: $GPRMC,123519,A,4807.038,N,01131.000,E,022.4,084.4,230394,003.1,W*6A
Sentence parsed correctly: $GPRMC,123519,A,4807.038,S,01131.000,E,022.4,084.4,230394,003.1,W*77
Sentence parsed correctly: $GPRMC,181704,A,5209.6815,N,00643.0724,E,000.1,187.0,310109,,*17
Sentence parsed correctly: $WIMWV,214.8,R,0.1,K,A*28
Sentence parsed correctly: $IIMWV,273,R,24.3,N,A*10
Sentence parsed correctly: $GPRMC,123519,A,4807.038,S,02121.000,W,015.4,084.4,230394,003.1,foobar,bazbladata*9
Sentence parsed correctly: $FOO,123,bar,bla,baz*11
Data overflow error for sentence.
Data overflow error for sentence.
Incorrect checksum. Calculated checksum: 0x6A, checksum received in sentence: 0x6B.
$GPRMC,123519,A,4807.038,N,01131.000,E,022.4,084.4,230394,003.1,W*6A
Summary:
 Bytes: 1165
 Errors: 3
 Sentences: 7
